### PR TITLE
Cancel subscription, donation cache, minor account page edits

### PR DESCRIPTION
**Pull request by jonbarrow**
Adds the endpoint to cancel a subscription, adds a cache to track donation progress, and some minor account page edits to show some ideas (need styling updates)

### CODE REVIEW

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/routers/account.js [hunk 1]
```diff
@@ -626,8 +669,8 @@ router.post('/checkout/:priceId', async (request, response) => {
 	const pnid = await database.PNID.findOne({ pid });
 
 	if (pnid.get('access_level') >= 2) {
-		response.cookie('error', 'Staff members do not need to purchase tiers', { domain: '.pretendo.network' });
-		return response.redirect('/account');
+		//response.cookie('error', 'Staff members do not need to purchase tiers', { domain: '.pretendo.network' });
+		//return response.redirect('/account');
 	}
```
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/routers/account.js:674]: Oops that’s a mistake on my part!

File: src/routers/account.js [hunk 2]
```diff
@@ -653,7 +696,80 @@ router.post('/checkout/:priceId', async (request, response) => {
...
+	}
+
+	// If still failed, something went horribly wrong
+	if (apiResponse.statusCode !== 200) {
+		return response.redirect('/account/login');
+	}
+
+	// Set user account info to render data
+	const account = apiResponse.body;
+	const pid = account.pid;
```
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/routers/account.js:739]: Same response as before

File: src/routers/account.js [hunk 3]
```diff
@@ -653,7 +696,80 @@ router.post('/checkout/:priceId', async (request, response) => {
...
+			if (pnid.get('access_level') < 2) {
+				// Fail-safe for if staff members reach here
+				// Mostly only useful during testing
+				updateData.access_level = 0;
+			}
+
+			await database.PNID.updateOne({ pid }, { $set: updateData }).exec();
+		} catch (error) {
+			logger.error(`Error canceling old user subscription | ${pnid.get('connections.stripe.customer_id')}, ${pid}, ${subscriptionId} | - ${error.message}`);
+		}
```
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/routers/account.js:765]: Will fix when I get home!

File: src/routers/account.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/routers/account.js]: Fixed

File: src/routers/account.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonbarrow) [reply src/routers/account.js]: Fixed

**Pull request merged**